packages/job-worker/src/playout/bucketAdlibJobs.ts

```diff
@@ -0,0 +1,94 @@
+import { ExecuteActionResult, ExecuteBucketAdLibOrActionProps } from '@sofie-automation/corelib/dist/worker/studio'
+import { JobContext } from '../jobs'
+import { runJobWithPlaylistLock } from './lock'
+import { UserError, UserErrorMessage } from '@sofie-automation/corelib/dist/error'
+import { BucketId, ShowStyleBaseId, ShowStyleVariantId } from '@sofie-automation/corelib/dist/dataModel/Ids'
+import { innerStartOrQueueAdLibPiece } from './adlibUtils'
+import { executeAdlibAction } from './adlibAction'
+import { RundownHoldState } from '@sofie-automation/corelib/dist/dataModel/RundownPlaylist'
+import { createPlayoutModelfromInitModel, loadPlayoutModelPreInit } from './model/implementation/LoadPlayoutModel'
+
+/**
+ * Execute an AdLib Action
+ */
+export async function handleExecuteBucketAdLibOrAction(
+	context: JobContext,
+	data: ExecuteBucketAdLibOrActionProps
+): Promise<ExecuteActionResult> {
+	return runJobWithPlaylistLock(context, data, async (playlist, lock) => {
+		// First load the minimum amount of data required to run the executeDataStoreAction handler
+		if (!playlist) throw new Error(`Job playlist "${data.playlistId}" not found `)
+
+		if (!playlist.activationId) throw UserError.create(UserErrorMessage.InactiveRundown, undefined, 412)
+		if (!playlist.currentPartInfo) throw UserError.create(UserErrorMessage.NoCurrentPart, undefined, 412)
+
+		const initCache = await loadPlayoutModelPreInit(context, lock, playlist, false)
+
+		const rundown = initCache.getRundown(playlist.currentPartInfo.rundownId)
+		if (!rundown) throw new Error(`Current Rundown "${playlist.currentPartInfo.rundownId}" could not be found`)
+
+		const { bucketAdLib, bucketAdLibAction } = await findBucketAdLibOrAction(
+			context,
+			data.bucketId,
+			data.externalId,
+			rundown.showStyleBaseId,
+			rundown.showStyleVariantId
+		)
+
+		if (bucketAdLib) {
+			if (playlist.holdState === RundownHoldState.ACTIVE || playlist.holdState === RundownHoldState.PENDING) {
+				throw UserError.create(UserErrorMessage.DuringHold)
+			}
+
+			const playoutModel = await createPlayoutModelfromInitModel(context, initCache)
```

Comment:
I don't think this ever gets saved back to mongo?

Reply:
I added a call saving the model at the end